Resources/views/CategoryAdmin/tree.html.twig

```diff
@@ -19,20 +19,22 @@ file that was distributed with this source code.
 {% macro navigate_child(collection, admin, root, depth) %}
     <ul{% if root %} class="sonata-tree sonata-tree--toggleable js-treeview"{% endif %}>
         {% for element in collection %}
-            <li class="sonata-ba-list-field" objectId="{{ element.id }}" >
```

Comment:
Indeed! You have directed me to another solution. There should be no bad elements in collection at all. Moved this check before calling macro.